File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mpp = {}
        for i in range(len(nums)):
            if target - nums[i] in mpp:
                if i < mpp[target-nums[i]]:
                    return [i, mpp[target-nums[i]]]
                else:
                    return [mpp[target-nums[i]], i]
            mpp[nums[i]] = i

File: Data Structures & Algorithms/combination-target-sum/submission-0.cpp
class Solution {
public:
    void solve(vector<int>& nums,int index,int remaining,vector<int>& current,vector<vector<int>>& ans){
        if(remaining == 0){
            ans.push_back(current);
            return;
        }
        if(remaining < 0 || index == nums.size()){
            return;
        }

        if(nums[index] <= remaining){
            current.push_back(nums[index]);
            solve(nums,index,remaining-nums[index],current,ans);
            current.pop_back();
        }

        solve(nums,index+1,remaining,current,ans);
    }
    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        vector<vector<int>> ans;
        vector<int> current;
        solve(nums,0,target,current,ans);
        return ans;
    }
};
